### PR TITLE
improve: add explicit tests for `getSelect()`, `getExpand()` and `properties-util.ts`

### PR DESCRIPTION
**Pull request by shojisoma**
Closes SAP/cloud-sdk-backlog#463.

this adds explicit tests for `getSelect()`, `getExpand()` and also `properties-util.ts`

### CODE REVIEW

File: packages/odata-common/src/properties-util.spec.ts [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
-import { isNavigationProperty } from './properties-util';
+import {
+  CommonEntity,
+  commonEntityApi
+} from '@sap-cloud-sdk/test-services-odata-common/common-entity';
+import { nonEnumerable, isNavigationProperty } from './properties-util';
 
 describe('isNavigationProperty', () => {
+  it('returns false for non-enumerable property', () => {
+    nonEnumerable(CommonEntity, '_entityName');
```
Comment (FrankEssenberger) [inline packages/odata-common/src/properties-util.spec.ts:9]: [req] this test should not be under the `isNavigationProperty` description.

File: packages/odata-v2/src/properties-util.spec.ts [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+  });
+
+  it('returns true for one to many navigation property ', () => {
     expect(isNavigationProperty('toMultiLink', testEntityApi.schema)).toBe(
       true
     );
   });
+
+  it('returns false for string property', () => {
+    expect(isNavigationProperty('stringProperty', testEntityApi.schema)).toBe(
```
Comment (FrankEssenberger) [inline packages/odata-v2/src/properties-util.spec.ts:32]: I guess this is already tested via the common package?

File: packages/odata-v2/src/properties-util.spec.ts [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
-    const { testEntityApi } = testService();
+  it('returns false for non-enumerable property', () => {
+    nonEnumerable(TestEntity, '_entityName');
+    expect(
+      Object.getOwnPropertyDescriptor(TestEntity, '_entityName')?.enumerable
+    ).toBeFalsy();
+  });
+
+  it('returns true for one to one navigation property ', () => {
+    expect(isNavigationProperty('toSingleLink', testEntityApi.schema)).toBe(
```
Comment (FrankEssenberger) [inline packages/odata-v2/src/properties-util.spec.ts:20]: ss this is already tested via the common package?

File: packages/odata-v4/src/properties-util.spec.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  it('returns true for one to many navigation property ', () => {
+    expect(isNavigationProperty('toMultiLink', testEntityApi.schema)).toBe(
+      true
+    );
+  });
+
+  it('returns false for string property', () => {
+    expect(isNavigationProperty('stringProperty', testEntityApi.schema)).toBe(
```
Comment (FrankEssenberger) [inline packages/odata-v4/src/properties-util.spec.ts:32]: ss this is already tested via the common package?

**Review by FrankEssenberger — APPROVED** [review]
LGTM - Perhaps you can remove a few of the util tests, because you have tested it already in the common package. But I am not sure. If you think we test something more leave then in.

File: packages/odata-v4/src/properties-util.spec.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import {
+  TestEntity,
+  testService
+} from '@sap-cloud-sdk/test-services-odata-v4/test-service';
+import {
+  nonEnumerable,
+  isNavigationProperty
+} from '@sap-cloud-sdk/odata-common/dist/properties-util';
```
Comment (deekshas8) [inline packages/odata-v4/src/properties-util.spec.ts:8]:
```suggestion
} from '@sap-cloud-sdk/odata-common/internal';
```

**Review by deekshas8 — CHANGES_REQUESTED** [review]

File: packages/odata-common/src/properties-util.spec.ts (hunk 1 above)
Comment (shojisoma) [reply packages/odata-common/src/properties-util.spec.ts:9]: I made another description for the test

File: packages/odata-v2/src/properties-util.spec.ts (hunk 2 above)
Comment (shojisoma) [reply packages/odata-v2/src/properties-util.spec.ts]: I removed this test

File: packages/odata-v2/src/uri-conversion/get-expand.spec.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import {
+  testEntityApi,
+  testEntityMultiLinkApi,
+  testEntitySingleLinkApi
+} from '../../test/test-util';
+import { getExpand } from './get-expand';
+
+describe('get expand', () => {
+  it('for first level expand without sub-query', () => {
```
[the remark this answers is not in the document]
Comment (shojisoma) [reply packages/odata-v2/src/uri-conversion/get-expand.spec.ts]: I made them complete sentences

File: packages/odata-v4/src/uri-conversion/get-expand.spec.ts [hunk 7]
```diff
@@ -7,7 +7,7 @@ import {
 import { getExpand } from './get-expand';
 
 describe('get expand', () => {
-  it('for first level expand without sub-query', () => {
+  it('should return all expand quey parameters', () => {
```
Comment (deekshas8) [inline packages/odata-v4/src/uri-conversion/get-expand.spec.ts:10]:
```suggestion
  it('should return all expand query parameters', () => {
```

**Review by deekshas8 — APPROVED** [review]
LGTM. Please adjust the typo.

**Pull request merged**